src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -441,6 +443,27 @@ export class MongoStorageAdapter {
     return Promise.resolve();
   }
 
+  createTextIndexesIfNeeded(className, query) {
+    for(const fieldName in query) {
+      if (!query[fieldName] || !query[fieldName].$text) {
+        continue;
+      }
+      const index = {
+        [fieldName]: 'text'
+      };
+      return this.createIndex(className, index)
+        .catch((error) => {
+          if (error.code === 85) {
```

Comment:
Instead of throwing error I could delete all current text indexes, add the new one and retry.